catalogue/webapp/pages/item.js

```diff
@@ -5,13 +5,113 @@ import fetch from 'isomorphic-unfetch';
 import { type IIIFManifest, type IIIFCanvas } from '@weco/common/model/iiif';
 import { iiifImageTemplate } from '@weco/common/utils/convert-image-uri';
 import { itemUrl, workUrl } from '@weco/common/services/catalogue/urls';
-import Layout12 from '@weco/common/views/components/Layout12/Layout12';
 import PageLayout from '@weco/common/views/components/PageLayout/PageLayout';
 import Paginator, {
   type PaginatorRenderFunctionProps,
 } from '@weco/common/views/components/RenderlessPaginator/RenderlessPaginator';
 import Control from '@weco/common/views/components/Buttons/Control/Control';
 import { classNames, spacing, font } from '@weco/common/utils/classnames';
+import styled from 'styled-components';
+import Layout12 from '@weco/common/views/components/Layout12/Layout12';
+
+const IIIFViewer = styled.div`
+  width: 100vw;
+  height: 100vh;
+  flex-direction: row-reverse;
+
+  .iiif-viewer__main {
+    height: 80%;
+    width: 100%;
+
+    @media (min-width: ${props => props.theme.sizes.medium}px) {
+      height: 100%;
+      width: 75%;
+    }
+  }
+
+  .iiif-viewer__thumbs {
+    height: 20%;
+    width: 100%;
+    padding: 0 100px 0 0;
+
+    @media (min-width: ${props => props.theme.sizes.medium}px) {
+      height: 100%;
+      flex-direction: column;
+      width: 25%;
+      padding: 0 0 100px;
+    }
+
+    .iiif-viewer__paginator-buttons {
+      left: auto;
+      right: ${props => props.theme.spacingUnit}px;
+      bottom: 50%;
+      transform: translateX(0%) translateY(50%);
+
+      @media (min-width: ${props => props.theme.sizes.medium}px) {
+        bottom: ${props => props.theme.spacingUnit * 5}px;
+        left: 50%;
+        transform: translateX(-50%) translateY(0%) rotate(90deg);
+      }
+    }
+  }
+
+  .iiif-viewer__thumb {
+    height: 100%;
+    width: 20%;
+    margin-right: ${props => props.theme.spacingUnit}px;
+
+    &:last-child {
+      margin: 0;
+    }
+
+    @media (min-width: 600px) {
+      height: 25%;
+      width: 100%;
+      margin-right: 0;
+    }
+  }
+
+  .iiif-viewer__thumb-number {
+    top: ${props => props.theme.spacingUnit}px;
+    left: 50%;
+    transform: translateX(-50%);
+    padding: 3px 2px 0;
+  }
+
+  .iiif-viewer__main-x-of-y {
+    top: ${props => props.theme.spacingUnit}px;
+    left: 50%;
+    transform: translateX(-50%);
+  }
+
+  .iiif-viewer__paginator-buttons {
+    bottom: 6px;
+    left: 50%;
+    transform: translateX(-50%);
+  }
+
+  .iiif-viewer__thumb-link {
+    height: 100%;
+
+    img {
+      border: 3px solid transparent;
+      transition: border-color 200ms ease;
+    }
+
+    &.is-active img {
+      border-color: ${props => props.theme.colors.white};
+    }
+  }
+
+  img {
+    margin: 0 auto;
+    display: block;
+    width: auto;
+    height: auto;
+    max-width: 100%;
+    max-height: 100%;
+  }
+`;
```

Comment:
Question (in no way a suggestion, just trying to make sure we get this right).
We seem to be mixing two paradigms here of SASS and styled components (SC).

To only use SC we could have something like:
```jsx
const IIIFViewer = styled.div`
  width: 100vw;
  height: 100vh;
  flex-direction: row-reverse;
`;

const IIIFViewerMain = styled.div`
  height: 80%;
  width: 100%;
  @media (min-width: ${props => props.theme.sizes.medium}px) {
    height: 100%;
    width: 75%;
  }
`;
//... etc

<IIIFViewer>
  <IIIFViewerMain><p>Oh hi<p/><IIIFViewerMain>
</IIIFViewer>
```

This seems to be what I can see from the examples, and I can't find examples of nesting plain SASS/CSS in the styled template string, and using it elsewhere.

Something like this, in my head, avoids two paradigms, and also avoids the specificity problem mentioned earlier.

As the components won't be exported, they can't be used elsewhere either, so just makes for tidy reading.

Thoughts?

Reply:
Thinking about, part of the problem was adding our root classNames in a nice way, just seen you can do this: https://www.styled-components.com/docs/faqs#can-i-use-css-frameworks

Reply:
I'd be OK doing it like that, yeah. The thing that's causing me to flip-flop opinion more is when/how we combine our root-scope classes with styled-components.

Reply:
If we have a root style class for what we use, we use it?

Reply:
## Non root style classes

```jsx
const Component = styled.div`
  background: special-colour-out-of-pallet;
  display: absolute;
`;

<Component>Hi</Component>

const Component2 = styled.div`
  background: another-special-colour-out-of-pallet;
  display: absolute;
`;

<Component2>Hi again</Component2>
```

```css
.style-Component1 {
  background: special-colour-out-of-pallet;
  position: absolute;
}

.style-Component2 {
  background: another-special-colour-out-of-pallet;
  position: absolute;
}
```

## Root style classes

```jsx
const Component = styled.div.attrs({ className: 'absolute' })`
  background: special-colour-out-of-pallet;
`;

<Component>Hi</Component>

const Component2 = styled.div.attrs({ className: 'absolute' })`
  background: another-special-colour-out-of-pallet;
`;

<Component2>Hi again</Component2>
```

```css
.style-Component1 {
  background: special-colour-out-of-pallet;
}

.style-Component2 {
  background: another-special-colour-out-of-pallet;
}

.absolute { position: absolute; }
```

But now apply this to all the spacing classes, fonts etc etc, and we land up creating tonnes more CSS that required.

Reply:
Saying this, I am completely bikeshedding this feature, and we can raise this in another PRif we want to get the feature out first.

Reply:
Definitely one for thrashing out in _Chewing the Code_.